src/avram/base_query_template.cr

```diff
@@ -112,13 +112,14 @@ class Avram::BaseQueryTemplate
             {% elsif assoc[:through] %}
               {{ join_type.downcase.id }}_join_{{ assoc[:through].id }}
                 .__yield_where_{{ assoc[:through].id }} do |join_query|
-                  join_query.{{ join_type.downcase.id }}_join_{{ assoc[:table_name] }}
+                  {% singular_association_name = run("../run_macros/singularize_word.cr", assoc[:assoc_name]) %}
```

Comment:
This needs to be conditional. Only associations where the "through" class has a "belongs_to" or a "has_one" association should be singularized, otherwise we should use the association name

We will also have to add in an optional property to specify the association name when it's something totally different.